modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx

```diff
@@ -275,6 +276,65 @@ const ActionForm = ({
 	const urlFormElementId = `${namespace}URL`;
 	const modalSizeFormElementId = `${namespace}ModalSize`;
 
+	const ModalBody = ({closeModal}: {closeModal: Function}) => {
+		const [filteredIconSymbols, setFilteredIconSymbols] = useState<
+			Array<{label: string; value: string}>
+		>([]);
+		const [query, setQuery] = useState('');
+
+		useEffect(() => {
+			setFilteredIconSymbols(availableIconSymbols);
+		}, []);
```

Comment:
Do we need for `useEffect` here? Can we can set initial value in `useState`? Something like 
```js
const [filteredIconSymbols, setFilteredIconSymbols] = useState<
	Array<{label: string; value: string}>
>(availableIconSymbols);
```

This would fix flickering behavior as you open the modal, as `useEffect` results in a re-render, where first one is empty modal:


https://github.com/liferay-frontend/liferay-portal/assets/5435511/75e42df2-638d-4c6c-bc9e-b81e961bf8ab

Reply:
Yeah, much better getting rid of the useEffect!